pkg/node/types/node.go

```diff
@@ -615,7 +616,23 @@ func (n *Node) Unmarshal(_ string, data []byte) error {
 		return err
 	}
 
+	if err := newNode.validate(); err != nil {
+		return err
+	}
+
 	*n = newNode
 
 	return nil
 }
+
+func (n *Node) validate() error {
+	// Skip the ClusterID check if it matches the local one, as we assume that
+	// it has already been validated, and to allow it to be zero.
+	if n.ClusterID != option.Config.ClusterID {
+		if err := cmtypes.ValidateClusterID(n.ClusterID); err != nil {
+			return err
+		}
+	}
+
+	return nil
```

Comment:
Possible style improvement:

```suggestion
	if n.ClusterID == option.Config.ClusterID {
		return nil
	}

	return cmtypes.ValidateClusterID(n.ClusterID)
```

Reply:
Generally speaking, I agree that your proposal is cleaner. I'd personally prefer to preserve the current approach, though, as it simplifies the addition of further checks, and it is consistent with the services validation.